src/frontend/src/optimizer/plan_node/generic/log_scan.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use educe::Educe;
+use pretty_xmlish::Pretty;
+use risingwave_common::catalog::{ColumnDesc, Field, Schema, TableDesc};
+use risingwave_common::types::DataType;
+use risingwave_common::util::sort_util::ColumnOrder;
+
+use crate::catalog::ColumnId;
+use crate::expr::{ExprRewriter, ExprVisitor};
+use crate::optimizer::optimizer_context::OptimizerContextRef;
+
+const OP_NAME: &str = "op";
+const OP_TYPE: DataType = DataType::Int16;
+
+#[derive(Debug, Clone, Educe)]
+#[educe(PartialEq, Eq, Hash)]
+pub struct LogScan {
+    pub table_name: String,
+    /// Include `output_col_idx` and `op_column`
+    pub output_col_idx: Vec<usize>,
+    /// Descriptor of the table
+    pub table_desc: Rc<TableDesc>,
+    /// Help `RowSeqLogScan` executor use a better chunk size
+    pub chunk_size: Option<u32>,
+
+    #[educe(PartialEq(ignore))]
+    #[educe(Hash(ignore))]
+    pub ctx: OptimizerContextRef,
+
+    pub old_epoch: u64,
+    pub new_epoch: u64,
+}
+
+impl LogScan {
+    pub fn rewrite_exprs(&self, _rewriter: &mut dyn ExprRewriter) {}
+
+    pub fn visit_exprs(&self, _v: &mut dyn ExprVisitor) {}
+
+    // Used for create batch exec, without op
+    pub fn output_column_ids(&self) -> Vec<ColumnId> {
+        self.output_col_idx
+            .iter()
+            .map(|i| self.get_table_columns()[*i].column_id)
+            .collect()
+    }
+
+    pub fn primary_key(&self) -> &[ColumnOrder] {
+        &self.table_desc.pk
+    }
+
+    pub(crate) fn column_names_with_table_prefix(&self) -> Vec<String> {
```

Comment:
Both `column_names_with_table_prefix` and `column_names` are only used internally. May remove the `pub(crate)` on them.